programs/monaco_protocol/src/instructions/order/cancel_order_post_market_lock.rs

```diff
@@ -0,0 +1,291 @@
+use anchor_lang::prelude::*;
+
+use crate::error::CoreError;
+use crate::instructions::clock::current_timestamp;
+use crate::instructions::market_position;
+use crate::state::market_account::{Market, MarketOrderBehaviour, MarketStatus};
+use crate::state::market_position_account::MarketPosition;
+use crate::state::order_account::Order;
+use crate::state::order_account::OrderStatus;
+
+pub fn cancel_order_post_market_lock(
+    market: &Market,
+    order: &mut Order,
+    market_position: &mut MarketPosition,
+) -> Result<u64> {
+    // market is open + should be locked and cancellation is the intended behaviour
+    require!(
+        [MarketStatus::Open].contains(&market.market_status),
+        CoreError::CancelationMarketStatusInvalid
+    );
+    require!(
+        market.market_lock_timestamp <= current_timestamp(),
+        CoreError::CancelationMarketNotLocked
+    );
+    require!(
+        MarketOrderBehaviour::CancelUnmatched.eq(&market.market_lock_order_behaviour),
+        CoreError::CancelationMarketOrderBehaviourInvalid
+    );
+
+    // TODO check that there are no order requests in the queue nor matches in the queue yet to be processed
+
+    // order is (open or matched) + there is remaining stake to be refunded
+    require!(
+        [OrderStatus::Open, OrderStatus::Matched].contains(&order.order_status),
+        CoreError::CancelationOrderStatusInvalid
+    );
+    require!(
+        order.stake_unmatched > 0_u64,
+        CoreError::CancelOrderNotCancellable
+    );
+
+    order.void_stake_unmatched(); // <-- void needs to happen before refund calculation
+    let refund = market_position::update_on_order_cancellation(market_position, order)?;
+
```

Comment:
I think now we need this here as well
```suggestion
    // if never matched
    if order.stake == order.voided_stake {
        // no more settlement needed
        market.decrement_unsettled_accounts_count()?;
    }

```

Reply:
This should be addressed now.